### PR TITLE
fix: resolve references in schema

### PR DESCRIPTION
**Pull request by sokra**
avoid accessing definitions directly

**What kind of change does this PR introduce?**
bugfix

**Did you add tests for your changes?**
existing test should be fine

**If relevant, did you update the documentation?**
not relevant

**Summary**
refactor the way of accessing webpack's schema to not create a internal dependency on the exact structure of the schema.

**Does this PR introduce a breaking change?**
No

**Other information**
see also https://github.com/webpack/webpack/pull/7232

### CODE REVIEW

**Comment by ev1stensberg** [comment]
@ematipico any reason for the fix prefix failing?

File: bin/config-yargs.js [hunk 1]
```diff
@@ -1,4 +1,58 @@
...
+		const path = schema.$ref.split("/");
+		for (const element of path) {
+			if (element === "#") {
+				current = optionsSchema;
+			} else {
+				current = current[element];
+			}
+		}
+	}
+	return schema;
```
Comment (sendilkumarn) [inline bin/config-yargs.js]: Is this `current`?

**Comment by ematipico** [comment]
> @ematipico any reason for the fix prefix failing?

Still broken, will fix in another PR

**Comment by dhruvdutt** [comment]
@ematipico Same CI issue is happening for #602 as well. Seems to be something related to `conventional-changelog-lint-config-cz`.

Could be happening because due to removal of `cz-conventional-changelog-lint` package.

**Comment by ematipico** [comment]
Yes, let's discuss it here #584

PR looks good to me!

**Pull request merged**